### PR TITLE
feat(saas): Send Organisation Invite via Email

### PR DESCRIPTION
**Pull request by DaniEnsi**
Adds the feature to automatically send invite email to people, instead of just providing a link to copy...

### CODE REVIEW

**Comment by DaniEnsi** [comment]
Hey @alifarooq9 ,
while using it, i wanted to try out the org invite stuff and though that it would be a nice touch if you could send invites straight from the app... If you don like it just close it :)

Looks like this:
image

File: starterkits/saas/src/app/(app)/(user)/org/members/invite/_components/send-invite-link.tsx [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+            }, 3000);
+        }
+
+        return () => {
+            clearTimeout(timeout);
+        };
+    }, [isSent]);
+
+    return (
+        <div className="flex gap-4 md:flex-row md:items-center md:justify-between">
```
Comment (alifarooq9) [inline starterkits/saas/src/app/(app)/(user)/org/members/invite/_components/send-invite-link.tsx:56]: Can you please use shadcn/ui form for better accessibility. for eg submit by pressing enter key, and zod email validation, currently I can send invite on non email input aswell for eg: "someword". refer others forms in codebase if you want some help or reach me out

File: starterkits/saas/src/server/actions/send-org-invite-email.ts
Comment (alifarooq9) [inline starterkits/saas/src/server/actions/send-org-invite-email.ts:1]: Use zod validation for email.

**Review by alifarooq9 — COMMENTED** [review]
@DaniEnsi thanks for adding the feature, I have reviewed it and added some comment. overall I like the approach just use validation and shadcn form, i have left a detail comment in file comments

**Comment by alifarooq9** [comment]
@DaniEnsi are you working on the changes I suggested or should I do it?

**Comment by DaniEnsi** [comment]
@alifarooq9 I think i did all you requested now, do you mind having a look?

**Comment by alifarooq9** [comment]
> @alifarooq9 I think i did all you requested now, do you mind having a look?

Getting this error

Screenshot 2024-04-12 at 12 50 05 PM

**Comment by DaniEnsi** [comment]
This is so weird, if I check that file in my pr (starterkits/saas/src/server/actions/send-org-invite-email.ts) it does not have the same code...

**Comment by DaniEnsi** [comment]
image

**Review by alifarooq9 — COMMENTED** [review]
@DaniEnsi Everything looks good I have slowed the issue of preview deployment and you can also see the preview that its working now. I just want you to add RESEND_API_KEY in your github action environment, I think thats why build check is failing.

Screenshot 2024-04-12 at 2 25 56 PM

**Comment by DaniEnsi** [comment]
@alifarooq9 I added the environment variable for my GitHub repo, but it has to be on my fork, right?

**Comment by DaniEnsi** [comment]
Also are you sure that I need to add the variable, as I can only add it to my forked repo, and on here it is your action that is run, not mine?

**Comment by alifarooq9** [comment]
> @alifarooq9 I added the environment variable for my GitHub repo, but it has to be on my fork, right?

Yeah, I think so

**Comment by alifarooq9** [comment]
> Also are you sure that I need to add the variable, as I can only add it to my forked repo, and on here it is your action that is run, not mine?

I already have my env variable set, and when I run checks on my PR it does not give me this error. maybe checks are running on your main because it's not merge yet, Idk I am just guessing

**Comment by DaniEnsi** [comment]
I mean I added the env to my main branch now, so it should work now

**Comment by alifarooq9** [comment]
> I mean I added the env to my main branch now, so it should work now

Again same error, I don't know why this is happening. Where did you put the secret?

**Comment by DaniEnsi** [comment]
Ok, I think it should work now, for me on my repo it wasn't working either the whole time when I ran the CI. But I added it to like all the options now so now it passes

**Comment by DaniEnsi** [comment]
IMG_0225

**Comment by DaniEnsi** [comment]
Ok, it failed again, now i have no clue…

**Comment by alifarooq9** [comment]
> Ok, it failed again, now i have no clue…

Same, let me try somethings

**Comment by DaniEnsi** [comment]
Nah its gonna fail again

**Comment by DaniEnsi** [comment]
IMG_0227

It passes nothing in

**Comment by DaniEnsi** [comment]
Check the answer with 19 upvotes:

https://stackoverflow.com/questions/58737785/github-actions-empty-env-secrets

**Comment by DaniEnsi** [comment]
@alifarooq9 Also have you tried, maybe adding a repository secret? That could maybe solve the issue

**Comment by alifarooq9** [comment]
> @alifarooq9 Also have you tried, maybe adding a repository secret? That could maybe solve the issue

I already added it, Leave it. Everything work's find I'll merge it

**Pull request merged**